controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -423,7 +425,18 @@ func (r *KubeadmControlPlaneReconciler) generateKubeadmConfig(ctx context.Contex
 	return bootstrapRef, nil
 }
 
+func (r *KubeadmControlPlaneReconciler) pickFailureDomain(ctx context.Context, kcp *controlplanev1.KubeadmControlPlane, cluster *clusterv1.Cluster) string {
```

Comment:
Similar comment here wrt naming, this is definitely the behavior that we want for scale up events, but we'd want the inverse behavior for scale down operations.